File: src/main/java/de/cmo/cache/domain/geschlecht/Geschlecht.java
package de.cmo.cache.domain.geschlecht;

import lombok.*;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
class Geschlecht {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String code;
    private String bezeichnung;
}

File: src/main/java/de/cmo/cache/domain/geschlecht/Geschlecht.kt
package de.cmo.cache.domain.geschlecht

import javax.persistence.*

@Entity
class Geschlecht {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null
    val code: String? = null
    val bezeichnung: String? = null

    @OneToMany(mappedBy = "geschlecht", cascade = [CascadeType.ALL], orphanRemoval = true)
    val geschlechtHists: List<GeschlechtHist>? = null
}